firebase-firestore/src/main/java/com/google/firebase/firestore/Query.java

```diff
@@ -481,110 +469,80 @@ private Value parseDocumentIdValue(Object documentIdValue) {
     }
   }
 
-  /** Validates that the value passed into a disjunctive filter satisfies all array requirements. */
-  private void validateDisjunctiveFilterElements(Object value, Operator op) {
-    if (!(value instanceof List) || ((List) value).size() == 0) {
-      throw new IllegalArgumentException(
-          "Invalid Query. A non-empty array is required for '" + op.toString() + "' filters.");
-    }
-    if (((List) value).size() > 10) {
-      throw new IllegalArgumentException(
-          "Invalid Query. '"
-              + op.toString()
-              + "' filters support a maximum of 10 elements in the value array.");
+  /**
+   * Parses the given filter's value from Object type to Value type.
+   *
+   * @return A new field filter that contains the parsed Value.
+   */
+  private FieldFilter parseFieldFilterValue(FieldFilter filter) {
```

Comment:
I'll try this suggestion.

Reply:
@schmidt-sebastian I think this wouldn't be a good idea because `FieldFilter` and `CompositeFilter` must inherit from `Filter`. If we add field and operator and value (as `Object`) in `com.google.firebase.fiterstore.Filter`, then `FieldFilter` and `CompositeFilter` will have those members too.

In this PR:
```
Filter Class (no members)
^    ^
|    |
|    FieldFilter Class. Members:
|    Field
|    Op
|    value (Value)
|    valueObject (Object)
|
CompositeFilter Class. Members:
Filter[]
logicType
```

If we keep `Object` in the API layer:
```
Filter Class. Members: Field, Op, valueObject (Object)
^    ^
|    |
|    FieldFilter Class. Members:
|    Field(by inheritance)
|    Op(by inheritance)
|    valueObject (Object) (by inheritance)
|    value (Value)
|
CompositeFilter Class. Members:
Field(by inheritance)
Op(by inheritance)
valueObject (Object) (by inheritance)
Filter[]
logicType
```

Another alternative is to create a third Filter class (E.g. `com.google.firebase.fiterstore.core.Filter`)
```
Filter Class (no members)
^   ^    ^
|   |    |
|   |    FieldFilter Class. Members:
|   |    Field
|   |    Op
|   |    value (Value)
|   | 
|   |
|   CompositeFilter Class. Members:
|   Filter[]
|   logicType
|
com.google.firebase.fiterstore.core.Filter Class. Members:
Field
Op
value (Object)
```
This is what I did initially (and sadly chose `UnqualifiedFilter` as the name instead of `core.Filter`).

What do you think? Did I misunderstand your suggestion?

Reply:
I think the Web API has a good precedent for this: 

The API layer only knows about the `Object`: https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/src/lite-api/query.ts#L132
This then gets translated to `Value` when we construct the internal Filter.

There is some duplication here (since you need to keep the filter operator) but the two classes are distinct from one another.

My main goal is to get the `Object` out of our core layer and into the API layer. If you add an hierarchy here that shares some members you might not achieve this goal.